### ISSUE TITLE
pyupgrade typing rewrites should not rewrite runtime types

### ISSUE THREAD

**Issue by JonathanPlasse** [charliermarsh/ruff#778]
This
```python
from __future__ import annotations
from typing import Optional, List

AnTypeAlias = List[Optional[float]]
```
should not be replaced by
```python
from __future__ import annotations

AnTypeAlias = list[float | None]
```
Python versions before 3.9 do not support type subscription, and 3.10 do not support pipe operator.

**Comment by charliermarsh** [comment]
Thanks! I’ll have to review the rules here and see how pyupgrade handles.

**Comment by andersk** [comment]
We correctly control this by the `target-version` setting.

```console
$ ruff --select=U --target-version=py38 test.py
$ ruff --select=U --target-version=py39 test.py
Found 1 error(s).
test.py:4:15: U006 Use `list` instead of `List` for type annotations
1 potentially fixable with the --fix option.
$ ruff --select=U --target-version=py310 test.py
Found 2 error(s).
test.py:4:15: U006 Use `list` instead of `List` for type annotations
test.py:4:20: U007 Use `X | Y` for type annotations
2 potentially fixable with the --fix option.
```

**Comment by charliermarsh** [comment]
Oh right -- this looks correct to me. I thought the issue had to do with using the standard library type annotations for assignment (on the RHS), rather than as annotations.

**Comment by charliermarsh** [comment]
Gonna close for now but feel free to ping if we're misinterpreting the problem!

**Comment by JonathanPlasse** [comment]
Maybe the default version should be the minimum supported Python version like pyupgrade does.
